autorest/codegen/templates/setup.py.jinja2

```diff
@@ -16,7 +15,13 @@ VERSION = "{{ code_model.options.get('package_version', '0.0.0') }}"
 # prerequisite: setuptools
 # http://pypi.python.org/pypi/setuptools
 
-REQUIRES = ["msrest>=0.6.21", "azure-core<2.0.0,>=1.20.1"{{ azure_mgmt_core_import }}]
+REQUIRES = [
+    "{{ code_model.package_dependency["dependency_msrest"] }}",
+    "{{ code_model.package_dependency["dependency_azure_core"] }}",
+    {%- if code_model.options["azure_arm"] %}
+    "{{ code_model.package_dependency["dependency_azure_mgmt_core"] }}"
```

Comment:
I think by adding the `-` in your if and endif, it's causing the generated code to format weirdly (you're telling it to get rid of whitespace when it doesn't need to. I think you can get rid of both, if not at least the one on endif